server/src/frames/frames.controller.ts

```diff
@@ -6,24 +6,54 @@ import {
   Param,
   Delete,
   UseGuards,
+  UseInterceptors,
+  UploadedFile,
+  ParseIntPipe,
 } from '@nestjs/common';
+import { FileInterceptor } from '@nestjs/platform-express';
 import { Prisma } from '@prisma/client';
 import { AuthGuard } from 'src/auth/auth.guard';
+import { FileService } from 'src/file/file.service';
 import { FramesService } from './frames.service';
 
 @Controller('frames')
 @UseGuards(AuthGuard)
 export class FramesController {
-  constructor(private readonly framesService: FramesService) {}
+  constructor(
+    private readonly framesService: FramesService,
+    private readonly fileService: FileService,
+  ) {}
 
   @Post()
-  create(@Body() data: Prisma.FrameCreateInput) {
-    return this.framesService.create(data);
+  @UseInterceptors(FileInterceptor('file'))
+  async create(
```

Comment:
I know you're using DTOs with the animation controller; might be worth keeping the approach consistent across the controllers.